### DISCUSSION TITLE
Adding an event listener to a form

### DISCUSSION THREAD

**Discussion by ManuelSagorski** [symfony/symfony#48266] [Q&A] [answered]
Hi together,

I have some problems to add an event listener to a form. I think I have done it as recommendet here: https://symfony.com/doc/6.0/form/dynamic_form_modification.html#adding-an-event-listener-to-a-form-class

This is how my code looks:
```
class AddVesselFormType extends AbstractType
{
...
    public function buildForm(FormBuilderInterface $builder, array $options): void
    {
      ...

      $builder->addEventListener(FormEvents::SUBMIT, function(FormEvent $event) {
          $submittedData = $event->getData();
      
          if(empty($submittedData['imo']) && empty($submittedData['eni'])) {
              throw new TransformationFailedException(sprintf(
                  'at least one field is required',
                  'imo, eni'
              ));
          }
      });
...
```

But I'm only getting this error:

App\Form\AddVesselFormType::App\Form\{closure}(): Argument #1 ($event) must be of type App\Form\FormEvent, Symfony\Component\Form\Event\SubmitEvent given, called in /var/www/symfony_docker/vendor/symfony/event-dispatcher/EventDispatcher.php on line 230

What am I doing wrong here?

**Comment by norkunas** [comment, marked as answer]
Did you add the import statement of FormEvent?

`use Symfony\Component\Form\FormEvent;`

**Reply by ManuelSagorski** [reply]
Oh no - that was the problem :-(
